libs/components/packages/src/schematics/migrations/update-9/ag-grid-29/ag-grid-29.schematic.ts

```diff
@@ -35,25 +35,39 @@ function removeAgGridCssFiles(): Rule {
   return async (tree: Tree): Promise<void> => {
     const workspace = await readWorkspace(tree);
     workspace.projects.forEach((project, projectName) => {
-      const build = project.targets.get('build');
-      if (build) {
-        build.options ??= {} as Record<string, JsonValue | undefined>;
-        build.options['styles'] ??= [] as string[];
-        build.options['styles'] = (build.options['styles'] as string[]).filter(
-          (cssFile) => !cssFile.startsWith('ag-grid-community/dist/styles/')
-        );
+      ['build', 'test'].forEach((targetName) => {
+        if (
+          targetName === 'build' &&
+          project.extensions['projectType'] === 'library'
+        ) {
+          return;
+        }
+        const targetDefinition = project.targets.get(targetName);
         if (
-          !build.options['styles'].includes(
-            'node_modules/@skyux/ag-grid/css/sky-ag-grid.css'
-          )
+          targetDefinition?.options &&
+          Array.isArray(targetDefinition.options['styles'])
         ) {
-          build.options['styles'].push(
-            'node_modules/@skyux/ag-grid/css/sky-ag-grid.css'
+          targetDefinition.options['styles'] = (
+            targetDefinition.options['styles'] as string[]
+          ).filter(
+            (cssFile) => !cssFile.startsWith('ag-grid-community/dist/styles/')
```

Comment:
Wouldn't this remove `ag-grid-community` stylesheets before checking if SKY UX/AG Grid stylesheets are being used? There's a slim chance, but one of our external consumers might have a `build` architect that uses the `ag-grid-community` stylesheet but doesn't use SKY UX components. In that case, we'd want to leave the `styles` alone. Maybe put this logic in the `if` block below?

Reply:
Good call. Updated.